src/iceberg/schema.cc

```diff
@@ -19,13 +19,57 @@
 
 #include "iceberg/schema.h"
 
+#include <algorithm>
 #include <format>
+#include <functional>
 
 #include "iceberg/type.h"
 #include "iceberg/util/formatter.h"  // IWYU pragma: keep
+#include "iceberg/util/macros.h"
+#include "iceberg/util/visit_type.h"
 
 namespace iceberg {
 
+class IdToFieldVisitor {
+ public:
+  explicit IdToFieldVisitor(
+      std::unordered_map<int32_t, std::reference_wrapper<const SchemaField>>&
+          id_to_field);
+  Status Visit(const Type& type);
+  Status VisitNestedType(const Type& type);
+
+ private:
+  std::unordered_map<int32_t, std::reference_wrapper<const SchemaField>>& id_to_field_;
+};
+
+class NameToIdVisitor {
```

Comment:
If read carefully, you might find that `IdToFieldVisitor` and `NameToIdVisitor` have used different styles of `Visit` functions. Should we make them consistent? For example, changing `IdToFieldVisitor` as below?

```cpp
class IdToFieldVisitor {
 public:
  Status Visit(const PrimitiveType& type);
  Status Visit(const NestedType& type);
};
```